To: vim_dev@googlegroups.com
Subject: Patch 7.3.460
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.460
Problem:    Win32: UPX does not compress 64 bit binaries.
Solution:   Mention and add the alternative: mpress. (Dave Bodenstab)
Files:	    src/INSTALLpc.txt, src/Make_ming.mak


*** ../vim-7.3.459/src/INSTALLpc.txt	2011-07-15 13:51:57.000000000 +0200
--- src/INSTALLpc.txt	2012-02-29 14:01:34.000000000 +0100
***************
*** 239,244 ****
--- 239,250 ----
  found at
      http://www.upx.org/
  
+ As of 2011, UPX still does not support compressing 64-bit EXE's; if you have
+ built a 64-bit vim then an alternative to UPX is 'MPRESS'. MPRESS can be found
+ at:
+     http://www.matcode.com/mpress.htm
+ 
+ 
  ADDITION: NLS support with MinGW
  
  (by Eduardo F. Amatria <eferna1@platea.pntic.mec.es>)
*** ../vim-7.3.459/src/Make_ming.mak	2011-10-12 14:11:43.000000000 +0200
--- src/Make_ming.mak	2012-02-29 16:54:09.000000000 +0100
***************
*** 14,22 ****
  # it's just run out of memory or something.  Run again, and it will continue
  # with 'xxd'.
  #
! # "make upx" makes *compressed* versions of the GUI and console EXEs, using the
! # excellent UPX compressor:
  #     http://upx.sourceforge.net/
  #
  # Maintained by Ron Aaron <ronaharon@yahoo.com>
  # updated 2003 Jan 20
--- 14,24 ----
  # it's just run out of memory or something.  Run again, and it will continue
  # with 'xxd'.
  #
! # "make upx" makes *compressed* versions of the 32 bit GUI and console EXEs,
! # using the excellent UPX compressor:
  #     http://upx.sourceforge.net/
+ # "make mpress" uses the MPRESS compressor for 32- and 64-bit EXEs:
+ #     http://www.matcode.com/mpress.htm
  #
  # Maintained by Ron Aaron <ronaharon@yahoo.com>
  # updated 2003 Jan 20
***************
*** 640,645 ****
--- 642,651 ----
  	upx gvim.exe
  	upx vim.exe
  
+ mpress: exes
+ 	mpress gvim.exe
+ 	mpress vim.exe
+ 
  xxd/xxd.exe: xxd/xxd.c
  	$(MAKE) -C xxd -f Make_ming.mak CC=$(CC)
  
*** ../vim-7.3.459/src/version.c	2012-02-29 13:58:43.000000000 +0100
--- src/version.c	2012-02-29 16:54:40.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     460,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
119. You are reading a book and look for the scroll bar to get to
     the next page.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
